### PR TITLE
P34: add, mult + Generic implementation

### PR DESCRIPTION
**Pull request by alvarogarcia7**

### CODE REVIEW

File: p34/src/biguint.rs [hunk 1]
```diff
@@ -0,0 +1,212 @@
...
+    pub fn from(initial_value: [u64; N]) -> Self {
+        Self {
+            value: initial_value,
+        }
+    }
+
+    pub fn new() -> Self {
+        Self::from([0; N])
+    }
+    pub fn sum(&mut self, another: &Self) {
```
Comment (newpavlov) [inline p34/src/biguint.rs]: You should implement the `sub::ops` trait instead of these inherent methods.

Comment (alvarogarcia7) [reply p34/src/biguint.rs]: I implemented `std::ops::{Add, Mul}`, left AddAssign for later

**Comment by alvarogarcia7** [comment]
@newpavlov could you please review this?

File: p34/tests/integration.rs [hunk 2]
```diff
@@ -0,0 +1,177 @@
...
+
+        assert_eq!(actual, ______expected);
+    }
+
+    #[test]
+    fn add_using_the_operator() {
+        let m = u64::MAX;
+        let mut _operand_1 = build_biguint_max_64!(m, m, 0, 0, 0, m, 0);
+        let _____operand_2 = build_biguint_max_64!(1, 0, 0, 0, 0, 1, 1);
+        let ______expected = build_biguint_max_64!(0, 0, 1, 0, 0, 0, 2);
```
Comment (newpavlov) [inline p34/tests/integration.rs]: Such naming is not idiomatic. Underscore is usually used to mark unused variables.

**Review by newpavlov — APPROVED** [review]

File: p34/tests/integration.rs (hunk 2 above)
Comment (alvarogarcia7) [reply p34/tests/integration.rs]: done

File: p34/src/biguint.rs [hunk 3]
```diff
@@ -0,0 +1,231 @@
...
+        BigUintGeneric::new();
+    };
+    // Define the first few values, everything else is zero
+    ($($value:expr $(,)?)+) => {
+        {
+            // let mut v: Vec<u64> = vec![];
+            // $(
+            // v.push($value);
+            // )*
+            // // TODO Question: by letting the `let variable` define the type, how may I get the length of the needed array?. How do I know if it's 64 or 92, etc?
```
Comment (alvarogarcia7) [inline p34/src/biguint.rs:212]: question here @newpavlov

Comment (newpavlov) [reply p34/src/biguint.rs:212]: The easiest way will be to do something like this:
```rust
let buf = [$($value,)*];
BigUintGeneric::from_slice(&buf)
```

**Pull request merged**